File: Python/392__Is_Subsequence.py
class Solution(object):
    def isSubsequence(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        if not s:
            return True
        s_ptr = 0
        for char in t:
            if char == s[s_ptr]:
                s_ptr += 1
                if s_ptr == len(s):
                    return True

        return False

File: Java/392__Is_Subsequence.java
class Solution {
    public boolean isSubsequence(String s, String t) {
        int n = s.length(), m = t.length();
        if (n == m)
            return s.equals(t);
        int index = -1;
        for (int i = 0; i < n; i++) {
            index = t.indexOf(s.charAt(i), index) + 1;
            if (index == 0)
                return false;
        }
        return true;
    }
}
